examples/SmartBank/test/SmartBankTest.js

```diff
@@ -0,0 +1,171 @@
+const Ae = require('@aeternity/aepp-sdk').Universal
+const Crypto = require('@aeternity/aepp-sdk').Crypto 
+
+const config = {
+  host: 'http://localhost:3001/',
+  internalHost: 'http://localhost:3001/internal/',
+  gas: 200000,
+  ttl: 55
+}
+
+async function deployContract (contractName, ...params) {
+  const [owner, gas, deployObj] = [params[0], params[1], params[2]]
+  const contractSource = utils.readFileRelative(
+    `./contracts/${contractName}.aes`,
+    'utf-8'
+  )
+  const compileContract = await owner.contractCompile(contractSource, gas)
+  const deployPromiseContract = await compileContract.deploy(deployObj)
+  return deployPromiseContract
+}
+
+async function callContract (contract, functionName, args, decodeType = 'int') {
+  const result = await contract.call(functionName, args)
+  const decodedResult = await result.decode(decodeType)
+  return decodedResult.value
+}
+function decodeContractAddress (key) {
+  const decoded58address = Crypto.decodeBase58Check(key.split('_')[1]).toString(
+    'hex'
+  )
+  return `0x${decoded58address}`
+}
+
+describe('SmartBank Contract', () => {
+  let owner
+  let SmartBankContract
+
+  before(async () => {
+    const ownerKeyPair = wallets[0]
+    owner = await Ae({
+      url: config.host,
+      internalUrl: config.internalHost,
+      keypair: ownerKeyPair,
+      nativeMode: true,
+      networkId: 'ae_devnet'
+    })
+  })
+
+  describe('Deploy contract', () => {
+    it('should deploy SmartBank contract', async () => {
+      const gas = { gas: config.gas }
+      const deployObj = {
+        options: { ttl: config.ttl }
+      }
+      SmartBankContract = await deployContract(
+        'SmartBank',
+        owner,
+        gas,
+        deployObj
+      )
+      assert(SmartBankContract.hasOwnProperty('address'))
+      assert(SmartBankContract.hasOwnProperty('owner'))
+    })
+  })
+
+  describe('Interact with the contract', () => {
+
+    it('should deposit tokens', async () => {
+      const args = {
+        args : '("12345")',
+        options: { ttl: 55 , amount: 1000},
+        abi: 'sophia'
+      }
+      const result = await callContract(
+        SmartBankContract,
+        'deposit',
+        args,
+        '()'
+      )
+      assert.deepEqual(result, [])
+
+    })
+    it('should check if user is registered in state', async () => {
+      const key = decodeContractAddress(wallets[0].publicKey)
+      const args = {
+        args : key, // pass the address of the caller
+        options: {ttl: 55},
+        abi: 'sophia'
+      }
+      wallets[0].publicKey
+      const result = await callContract(
+        SmartBankContract,
+        'user_exists',
+        args,
+        'bool'
+      )
+      assert.equal(result, '1')
+    })
```

Comment:
If the user does not exist, the code will `abort` the execution of the smart contract, but i don't think that there is some way of handling it in JS, as it will stop executing the contract.

Reply:
if this ```
  public function user_exists(user : address) : bool =
    if(Map.member(user, state))
      true
    else
      false```
 throw exception when user does not exists , maybe you should implement it with: 
```
switch(Map.lookup(key, some_map))
  None => false
  Some(x) => x
```

Reply:
The function itself returns a `boolean`, but this `boolean` is used in `require` function calls, which on the other hand, aborts the code execution if the `boolean` does not pass the check.

Reply:
okey, if it is aborted tested with assert.IsRejected(promise, errMsg) example: https://github.com/aeternity/aepp-sophia-examples/blob/master/libraries/MultisignatureWallet/test/multisig-wallet-tests.js#LC92

Reply:
Oh, i'm sorry, I think,  I misunderstood the question at the beginning. I will resolve this.